plugins/orchestrator-backend/src/service/SonataFlowService.test.ts

```diff
@@ -203,68 +187,57 @@ describe('SonataFlowService', () => {
       );
       expect(result).toEqual({ id: definitionId, status: 'completed' });
     });
-    it('should log an error and return undefined when the fetch response is not ok without extra info', async () => {
+    it('should propagate thrown error when the fetch response is not ok without extra info', async () => {
       // When
-      const mockResponse = setupTest({
+      setupTest({
         ok: false,
         status: 500,
         statusText: 'Internal Server Error',
         json: { details: undefined, stack: undefined },
       });
 
-      const result = await runErrorTest();
+      let result;
+      try {
+        await runErrorTest();
+      } catch (error) {
+        result = error;
+      }
 
-      // Then
-      expect(fetch).toHaveBeenCalledWith(
-        urlToFetch,
-        expectedFetchRequestInit(),
-      );
-      expect(result).toBeUndefined();
-      expect(loggerMock.error).toHaveBeenCalledTimes(1);
-      expect(loggerMock.error).toHaveBeenCalledWith(
-        `Error when executing workflow: Error: ${await sonataFlowService.createPrefixFetchErrorMessage(urlToFetch, mockResponse as Response, 'POST')}`,
-      );
+      expect(result).toBeDefined();
     });
-    it('should log an error and return undefined when the fetch response is not ok with extra info', async () => {
+    it('should propagate thrown exception when the fetch response is not ok with extra info', async () => {
       // When
-      const mockResponse = setupTest({
+      setupTest({
         ok: false,
         json: { details: 'Error details test', stack: 'Error stacktrace test' },
       });
 
-      const result = await runErrorTest();
+      let result;
+      try {
+        await runErrorTest();
+      } catch (error) {
+        result = error;
+      }
 
-      // Then
-      expect(fetch).toHaveBeenCalledWith(
```

Comment:
why removing this?

Reply:
expecting the fetch is absurd. we tell the mock to expect it and it will validate it . so why double test?